nucleus/iou_match.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import dataclass
+from enum import Enum
+
+from .constants import (
+    DATASET_ITEM_ID_KEY,
+    GROUND_TRUTH_ANNOTATION_ID_KEY,
+    GROUND_TRUTH_ANNOTATION_LABEL_KEY,
+    IOU_KEY,
+    MODEL_PREDICTION_ID_KEY,
+    MODEL_PREDICTION_LABEL_KEY,
+    MODEL_RUN_ID_KEY,
+)
+
+
+class ConfusionCategory(Enum):
+    TRUE_POSITIVE = "true_positive"
+    TRUE_NEGATIVE = "true_negative"
+    FALSE_POSITIVE = "false_positive"
+    FALSE_NEGATIVE = "false_negative"
+
+
+def infer_confusion_category(
+    ground_truth_annotation_label: str,
+    model_prediction_label: str,
+):
+    confusion_category = ConfusionCategory.FALSE_NEGATIVE
+
+    if (
+        model_prediction_label is None
+        and ground_truth_annotation_label is None
+    ):
+        confusion_category = ConfusionCategory.TRUE_NEGATIVE
+    elif model_prediction_label == ground_truth_annotation_label:
```

Comment:
```suggestion
    if model_prediction_label == ground_truth_annotation_label:
```
Never happens for our use cases. Let's just drop this.